### PR TITLE
Updated error handling mechanism

### LINKED ISSUE #136: ght issue on windows
getting a lot of errors running ght (on multipass on Windows)

- For this scenario, the output should be: `Invalid region, available regions are: americas, apac, emea`

### PR DESCRIPTION
**Pull request by ilayda21**
## DONE
- Updated error handling mechanism

## QA
- Checkout the feature branch
- Run `yarn run replicate <job-post-id> -r APAC`
- Verify an error which is human readable is displayed
- Run `yarn run replicate <job-post-id> -r apac`
- Verify job posts are created
- Run `yarn run replicate -i`
- Choose necessary options
- Verify job posts are created
-  Run `yarn run reset <job-id>`
- Verify job posts are deleted
-  Run `yarn run logout`
- Run `yarn run replicate -i`
- Verify it requires the credentials
-  Run `yarn run login`
- Enter credentials  
- Run `yarn run replicate -i`
- Verify it uses the saved credentials

## Fixes #136

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -99,273 +100,186 @@ async function deletePostsInteractive(
...
+        cloneFrom
+    );
+    // Mark all newly added job posts as live
+    const processedJobCount = await job.markAsLive(jobID, jobInfo.posts);
+
+    console.log(
+        green("✔"),
+        `${processedJobCount} job posts for ${clonedJobPosts
+            .map((post) => post.name)
+            .reduce(
```
Comment (goulinkh) [inline src/index.ts:166]: this should be replaced with `join` as reduce as I think `reduce` a complex function to read and `join` does the job very well in this case

File: src/index.ts [hunk 2]
```diff
@@ -449,18 +378,76 @@ async function main() {
...
+}
 
-    await program.parseAsync(process.argv);
+async function main() {
+    setupSentry();
+    const spinner = ora();
+    let browser: Browser | null = null;
+    try {
+        const program = new Command();
+        const sso = new SSO(spinner);
```
Comment (goulinkh) [inline src/index.ts:438]: When doing the SSO first, some commands like `help` & `logout`, `sso` will be required for them where it's in reality not required at all for the command `help` for example.

**Review by goulinkh — APPROVED** [review]
LGTM, added some tiny comments

File: src/index.ts (hunk 1 above)
Comment (ilayda21) [reply src/index.ts:166]: :+1:

File: src/index.ts (hunk 2 above)
Comment (ilayda21) [reply src/index.ts:440]: :+1:

**Pull request merged**